core/plugins/org.polarsys.capella.core.ui.semantic.browser/src/org/polarsys/capella/core/ui/semantic/browser/view/SemanticBrowserView.java

```diff
@@ -1120,10 +1135,11 @@ public void refresh(boolean forceRefresh) {
       // Precondition: do not set the same input twice, except during refreshing.
       TreeViewer currentTreeViewer = getCurrentViewer();
       Object lastInput = currentTreeViewer.getInput();
-      if (!forceRefresh && null != lastInput && lastInput.equals(input)) {
+      if (!forceRefresh && null != lastInput && lastInput.equals(input) && !hasOriginalSelectionChanged()) {
         return;
       }
 
+      this.previousOriginalSelection = this.currentOriginalSelection;
```

Comment:
maybe move it right aside L1148 `setInputOnViewers`

Reply:
Done